### PR TITLE
Fix #10284 - Add header_string option to writedlm()

### LINKED ISSUE #10284: Add header option to writedlm() and related
Sometimes we need to attach extra information to the data we're dumping:

``` julia
writecsv("foo.dat", data, header="This is the data...\nMeta info: ...")
```

Currently this is not possible.

### PR DESCRIPTION
**Pull request by juliohm**

### CODE REVIEW

**Comment by simonster** [comment]
Needs docs and a test

File: base/docs/helpdb.jl [hunk 1]
```diff
@@ -715,9 +715,9 @@ Raises a `SystemError` for `errno` with the descriptive string `sysfunc` if `ift
 systemerror
 
 doc"""
-    writedlm(f, A, delim='\\t')
+    writedlm(f, A, delim='\\t', header_string=nothing)
```
Comment (tkelman) [inline base/docs/helpdb.jl]: need to adjust the rst prototype as well, then run `./julia doc/genstdlib.jl`

generally rebases are preferred vs merges

**Comment by juliohm** [comment]
Thank you @tkelman, will do it.

**Comment by juliohm** [comment]
Ok, will close this pull request and start it over.

**Comment by tkelman** [comment]
That's not necessary, you can force push a rebased version to this same branch and it will overwrite what's here.

**Comment by juliohm** [comment]
You mean I can rebase my local master and push to GitHub and it will just work? Could you please elaborate?

**Comment by tkelman** [comment]
Yes.
```
git remote add JuliaLang https://github.com/JuliaLang/julia
git remote add juliohm https://github.com/juliohm/julia
git fetch --all
git checkout juliohm/master
git rebase -i JuliaLang/master
# don't select the merge commit
# make local changes, rebuild, regenerate docs, squash commits if you'd like
git push juliohm +master
```

**Comment by juliohm** [comment]
Thank you @tkelman, I'm having a problem to rebuild:

```
error: Your local changes to the following files would be overwritten by checkout:
	CMakeLists.txt
Please, commit your changes or stash them before you can switch branches.
Aborting
Makefile:2009: recipe for target '/home/julio/Projects/julia/deps/srccache/libgit2/CMakeLists.txt' failed
make[1]: *** [/home/julio/Projects/julia/deps/srccache/libgit2/CMakeLists.txt] Error 1
Makefile:78: recipe for target 'julia-deps' failed
make: *** [julia-deps] Error 2
```

I tried `make clenall` to make sure I'm starting clean, but it didn't help. Any help on how to proceed?

**Comment by tkelman** [comment]
sorry bout that, you can fix that one via `make -C deps distclean-libgit2`

**Comment by juliohm** [comment]
Now the last step, I tried:

```
git remote add juliossh git@github.com:juliohm/julia.git
git push juliossh +master
```

but it had no effect. Everything up-to-date. What is missing?

**Comment by tkelman** [comment]
You might have a local master that differs from the now-modified version of `juliohm/master`. Perhaps
```
git checkout master
git checkout -b master-old
git branch -D master
git checkout juliohm/master
git checkout -b master
git push juliossh +master
```
?

**Comment by juliohm** [comment]
The same result. Everything up-to-date. Do you think it is better to just open another PR with the same changes?

**Comment by tkelman** [comment]
Yeah, I guess so. Create a new branch (not master) at your fork for that one, just in case any rebasing is required.

**Comment by juliohm** [comment]
Thank you @tkelman , I will do that :+1:

**Pull request closed without merging**